### DISCUSSION TITLE
Add PatternFly Dropdown component...

### DISCUSSION THREAD

**Discussion by ddonahue007** [data-driven-forms/react-forms#1133] [Need help] [answered]
Hello, 
I have been using DDF on a project with the PFMapper. I am using the Select component that has already been ported to DDF, but I am having a hard time trying to figure out how change the width. By default select component's styles have it using 100% width, this looks a bit funky and not what I would like as the final product. Can I just pass properties?

**Comment by Hyperkid123** [comment]
@ddonahue007 it looks like we are missing the customization API for the select component. To explain a bit more, there is usually an API that allows customizing the composite component of a component. [You can check this component for example](https://data-driven-forms.org/provided-mappers/radio?mapper=mui#compositeprops).

This API is missing in the PF4 select component. We should add it.

**Comment by rvsia** [comment, marked as answer]
You can pass a `classname` in `FormGroupProps` object to change the width.

**Reply by Hyperkid123** [reply]
For that, I would recommend using the global props.

```JS
const componentMapper = {
  select: {
    component: Select,
    FormGroupProps: {
      className: 'foo',
    },
  },
};
```

**Reply by ddonahue007** [reply]
Thanks worked like a charm, I ended up dong the following:

```
export default {
  fields: [
    {
      component: 'select',
      label: 'Select',
      name: 'select',
      simpleValue: true,
      FormGroupProps: {
        style: { width: '30%' },
      },
      options: [
        {
          label: 'Dogs',
          value: '1',
        },
        {
          label: 'Cats',
          value: '2',
        },
        {
          label: 'Hamsters',
          value: '3',
        },
      ],
    },
  ],
};
```
